src/main/Plugins/ValidationPlugins/Dns/DnsValidation.cs

```diff
@@ -13,35 +14,59 @@ namespace PKISharp.WACS.Plugins.ValidationPlugins
     /// </summary>
     public abstract class DnsValidation<TOptions, TPlugin> : Validation<TOptions, Dns01ChallengeValidationDetails>
     {
-        public DnsValidation(ILogService logService, TOptions options, string identifier) : 
-            base(logService, options, identifier) { }
+        protected readonly DomainParser _domainParser;
+        protected readonly ILookupClientProvider _lookupClientProvider;
+
+        public DnsValidation(DomainParser domainParser, ILookupClientProvider lookupClientProvider, ILogService logService, TOptions options, string identifier) : 
+            base(logService, options, identifier)
+        {
+            _domainParser = domainParser;
+            _lookupClientProvider = lookupClientProvider;
+        }
 
         public override void PrepareChallenge()
         {
             CreateRecord(_challenge.DnsRecordName, _challenge.DnsRecordValue);
             _log.Information("Answer should now be available at {answerUri}", _challenge.DnsRecordName);
 
-            string foundValue = null;
             try
             {
-                if (IPAddress.TryParse(Properties.Settings.Default.DnsServer, out IPAddress address)) 
+                var domainName = _domainParser.Get(_challenge.DnsRecordName);
+                var lookupClient = _lookupClientProvider.Default;
+
+                _log.Debug("Querying SOA for {Domain}", domainName.RegistrableDomain);
+                var soaResponse = lookupClient.Query(domainName.RegistrableDomain, QueryType.SOA);
+
+                var nameServer = soaResponse.Answers.SoaRecords().FirstOrDefault()?.MName;
+                var nameServerIp = lookupClient.Query(nameServer, QueryType.A).Answers.ARecords().FirstOrDefault()?.Address;
+                _log.Debug("Registered Domain Name Server {DomainNameServer} ({DomainNameServerIp})", nameServer, nameServerIp);
+
+                var address = nameServerIp;
+
+                if (IPAddress.TryParse(Properties.Settings.Default.DnsServer, out IPAddress overrideNameServerIp))
```

Comment:
You noted that this might be a breaking change, and technically it is. However, since this a completely non-essential function and querying the actual name server(s) much better approaches what Let's Encrypt is doing, I would actually be in favor of removing the "DnsServer" setting. If we keep it around then I would not use it as an override, but rather as a fallback in case we cannot determine the registered name servers for whatever reason.

Reply:
well, i agree that it _should_ be unnecessary at this point but i don't think it hurts to keep the setting (empty) and allow it to be used. we can't assume how people might be using the setting. so honoring the setting as an override seems to me the best way to maintain backwards compatibility. that also minimizes the scope of the breaking change (which would be to remove `208.67.222.222`) from https://github.com/PKISharp/win-acme/blob/master/src/main/settings_default.config#L96.

i need you to make a decision on this one as it affects the implementation. ;)

Reply:
You've convinced me, let's keep it as an override!